proxy/InkAPI.cc

```diff
@@ -2137,7 +2137,8 @@ TSUrlLengthGet(TSMBuffer bufp, TSMLoc obj)
 char *
 TSUrlStringGet(TSMBuffer bufp, TSMLoc obj, int *length)
 {
-  sdk_assert(sdk_sanity_check_mbuffer(bufp) == TS_SUCCESS);
+  // bufp is not actually used anymore, so it can be null.
+  sdk_assert((sdk_sanity_check_mbuffer(bufp) == TS_SUCCESS) or (sdk_sanity_check_null_ptr(bufp) == TS_SUCCESS));
```

Comment:
I'd probably do the cheaper null check first.

Reply:
I'm a bit confused - `sdk_sanity_check_null_ptr` returns `TS_SUCCESS` if the argument is *not* `nullptr`. Therefore I don't think this allows a `nullptr`.

Reply:
You're looking for love in all the wrong places, review here:  https://github.com/apache/trafficserver/pull/3145

This PR is obsolete.